### PR TITLE
Fixes that the SA1412 fix all provider fixes violations with differen…

### LINKED ISSUE #1159: Equivalence key for SA1412 (StoreFilesAsUtf8) should depend on source encoding
Some encodings are not as easy to convert and require manual checking of special characters afterwards. This strategy will bring such files to the user's attention.

### PR DESCRIPTION
**Pull request by pdelvo**
…t encodings

Fixes #1159.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/MaintainabilityResources.resx [hunk 1]
```diff
@@ -133,6 +133,6 @@
     <value>Remove parentheses</value>
   </data>
   <data name="SA1412CodeFix" xml:space="preserve">
-    <value>Fix encoding</value>
+    <value>Change encoding from '{0}' to UTF-8 with byte order mark</value>
```
Comment (Noryoko) [inline StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/MaintainabilityResources.resx]: Should format stings be in resource files?

Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/MaintainabilityResources.resx]: Is there a reason not to?

Comment (Noryoko) [inline StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/MaintainabilityResources.resx]: Never mind. Looks consistent with our other code fix providers. :+1:

**Comment by Noryoko** [comment]
:+1:

File: StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1412FixAllProvider.cs [hunk 2]
```diff
@@ -36,34 +36,34 @@ internal sealed class SA1412FixAllProvider : FixAllProvider
...
-            var diagnostics = await fixAllContext.GetDocumentDiagnosticsAsync(document).ConfigureAwait(false);
+            Solution solution = project.Solution;
+
+            var documentIds = project.DocumentIds;
 
-            if (diagnostics.Length == 0)
+            foreach (var documentId in documentIds)
             {
-                return solution;
+                solution = await FixDocumentAsync(fixAllContext, solution.GetDocument(documentId)).ConfigureAwait(false);
```
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1412FixAllProvider.cs]: Could we process all documents in parallel via `Task.WhenAll`?

Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1412FixAllProvider.cs]: It's possible, but *believe me when I say* the problems with this code fix are not the performance of this block.

**Pull request merged**